microsoft/testsuites/cdrom/cdrom.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+import pathlib
+
+from assertpy import assert_that
+
+from lisa import Node, SkippedException, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.operating_system import Linux
+from lisa.sut_orchestrator import AZURE
+from lisa.testsuite import simple_requirement
+from lisa.tools import Gcc
+
+
+@TestSuiteMetadata(
+    area="cdrom",
+    category="functional",
+    description="""
+        Tests to check the behavior of the virtual cdrom device in VMs.
+    """,
+)
+class CdromSuite(TestSuite):
+    # CD drive status constants from linux uapi
+    # see https://github.com/torvalds/linux/blob/master/include/uapi/linux/cdrom.h#L417
+    _expected_device_status = (
+        "CDS_NO_DISC"  # drive tray is closed, no disk in the drive
+    )
+
+    @TestCaseMetadata(
+        description="""
+            Test to check the installation ISO is unloaded
+            after provisioning and rebooting a new VM.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[Linux], supported_platform_type=[AZURE]
+        ),
+    )
+    def verify_cdrom_device_status_code(self, node: Node) -> None:
+        # check there's a device to test
+        if not node.shell.exists(pathlib.PurePosixPath("/dev/cdrom")):
+            raise SkippedException(
+                "No /dev/cdrom device was present in this distro/image/vm."
+            )
+        # reboot to ensure there is no iso in the cd/dvd 'drive'
+        node.reboot()
+
+        # collect some paths before we start compiling...
+        local_path = pathlib.PurePath(__file__).parent.joinpath("cdstat.c")
+        working_path = node.get_working_path()
+        node_source_path = working_path.joinpath("cdstat.c")
+        node_exec_path = working_path.joinpath("cdstat")
+
+        # copy the tiny c program to the node (cdstat.c)
+        node.shell.copy(local_path=local_path, node_path=node_source_path)
+
+        # compile it
+        node.tools[Gcc].compile(
+            filename=str(node_source_path), output_name=str(node_exec_path)
+        )
+
+        # and run the test if the device is present.
+        result = node.execute(
+            cmd=str(node_exec_path),
+            shell=True,
+            sudo=True,
+        )
+        output = result.stdout.strip()
+
+        # first check for failure to open the cdrom device
+        assert_that(result.exit_code).described_as(
```

Comment:
I feel it doesn't need to assert for -1 separately. The raw output should be able to explain the error.

Reply:
I disagree, it's a subtle distinction but not being able to open /dev/cdrom implies a test bug (priviledges or distro differences) while the second assertion definitively implies a host bug. 

Having different error messages for each is a time saver when you're triaging through hundreds of tests. This is a bigger issue but there are a lot of error messages that LISA raises that I wish were more specific about _what should be done_ when you see the error raised.